src/components/ListPaginationComp/ListPagination.jsx

```diff
@@ -0,0 +1,102 @@
+import {
+  PaginationLi,
+  Paginationa,
+  PaginationUl,
+} from "./list-pagination-styles";
+import { getArticlesByPage } from "services/api-services";
+import PropTypes from "prop-types";
+
+/**
+ * This component take the number of articles and make a list of page numbers.
+ * 
+ * @param {Object}   props               An object of prop.
+ * @param {Number}   props.articlesCount The total number of articles.
+ * @param {Number}   props.currentPage   The current page.
+ * @param {Function} props.setArticles   A function that updates the state of articles.
+ * @param {Function} props.setPage       a function that updates the state of pageNumber.
+ */
+
+function ListPagination({ articlesCount, currentPage, setArticles, setPage }) {
+  if (articlesCount <= 10) {
+    return null;
+  }
+
+  /**
+   * Takes a page number then sets the page depending on the clicked list item (<PaginationLi>).
+   * Call a webapi to get articles by page number.
+   * Update the state by using setArticle prop after getting the response from the web api.
+   *
+   * @param {number} pageNo The number of the clicked PaginationLi.
+   */
+  const setpage = async (pageNo) => {
+    setPage(pageNo);
+    const response = await getArticlesByPage(pageNo);
+    setArticles(response.articles);
+  };
+
+  /**
+   *Call the set page function and give it the clicked list item (<PaginationLi>) number.
+   *
+   * @param {Number} pageNo The number of the current page.
+   */
+  const handleClick = (pageNo) => {
+    setpage(pageNo);
+  };
+
+  const range = [];
+  for (let i = 0; i < Math.ceil(articlesCount / 10); ++i) {
+    range.push(i);
+  }
+
+  return (
+    <PaginationUl className="list">
+      {range.map((pageNo) => {
+        const isCurrent = pageNo === currentPage;
+
+        return (
+          <PaginationLi
+            key={pageNo}
+            backgroundcolor={isCurrent ? true : false}
```

Comment:
i think this can be named better, maybe `isActive` also there is no need to store is current in a variable 
```suggestion
            iaActive={pageNo === currentPage}
```